docs/source/scenarios/vm_scheduling.rst

```diff
@@ -0,0 +1,368 @@
+* Average memory requested: 16.1 GB
+* Average CPU utilization: 15.6 %
+
+PM setting (Given by the /[topologies]/config.yml):
+
+* Amount: 880
+* CPU Cores: 16
+* Memory: 112 GB
+
+Naive Baseline
+^^^^^^^^^^^^^^^
+
+Belows are the final environment metrics of the method **Random Allocation** and 
+**Best-Fit Allocation** in different topologies. 
+For each experiment, we setup the environment and test for a duration of 30 days.
+Besides, we use several settings of PM capacity to test performance under different 
+initial resources.
+
+
+Random Allocation
+~~~~~~~~~~~~~~~~~~~~
+
+Randomly allocate to a valid PM.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Topology
+     - PM Setting
+     - Total VM Requests
+     - Total Energy Consumption
+     - Successful Allocation
+     - Successful completion
+     - Failed Allocation
+   * - Azure.2019.10k 
+     - 100 PMs, 32 Cores, 128 GB
+     - 10,000
+     - 2,430,651.6
+     - 9,850
+     - 9,030
+     - 150
+   * - 
+     - 100 PMs, 16 Cores, 112 GB
+     - 10,000
+     - 2,978,445.0
+     - 8,011
+     - 7,411
+     - 1,989
+   * - Azure.2019.336k
+     - 880 PMs, 32 Cores, 128 GB
+     - 335,985
+     - 26,681,249.7
+     - 176,468
+     - 165,715
+     - 159,517
+   * -  
+     - 880 PMs, 16 Cores, 112 GB
+     - 335,985
+     - 26,367,238.7
+     - 92,885
+     - 87,153
+     - 243,100
+
+Best-Fit Allocation
+~~~~~~~~~~~~~~~~~~~~
+
+Choose the valid PM with the least remaining resources (only consider CPU cores here).
+
+.. list-table::
+   :header-rows: 1
+
+   * - Topology
+     - PM Setting
+     - Total VM Requests
+     - Total Energy Consumption
+     - Successful Allocation
+     - Successful completion
+     - Failed Allocation
+   * - Azure.2019.10k 
+     - 100 PMs, 32 Cores, 128 GB
+     - 10,000
+     - 2,395,328.7
+     - 10,000
+     - 9,180
+     - 0
+   * - 
+     - 100 PMs, 16 Cores, 112 GB
+     - 10,000
+     - 2,987,086.6
+     - 7,917
+     - 7,313
+     - 2,083
+   * - Azure.2019.336k 
+     - 880 PMs, 32 Cores, 128 GB
+     - 335,985
+     - 26,695,470.8
+     - 171,044
+     - 160,495
+     - 164,941
+   * - 
+     - 880 PMs, 16 Cores, 112 GB
+     - 335,985
+     - 26,390,972.9
+     - 92,263
+     - 86,600
+     - 243,722
+
+
+
+Quick Start
+------------
+
+Data Preparation
+^^^^^^^^^^^^^^^^^
+
+When the environment is first created, the system will automatically trigger the pipeline to download 
+and process the data files. Afterwards, if you want to run multiple simulations, the system will detect
+whether the processed data files exist or not. If not, it will then trigger the pipeline again. Otherwise,
+the system will reuse the processed data files. 
+
+
+Environment Interface
+^^^^^^^^^^^^^^^^^^^^^^
+
+Before starting interaction with the environment, we need to know the definition of ``DecisionPayload`` and 
+``Action`` in VM Scheduling scenario first. Besides, you can query the environment snapshot list to get more 
```

Comment:
add link for snapshot list here

Reply:
https://maro.readthedocs.io/en/latest/key_components/data_model.html#advanced-features